src/future-hopr-lib-components/Layout/drawer.tsx

```diff
@@ -1,156 +1,147 @@
+    };
+  }, []);
 
-  const drawer = (
-    <div>
-      <AppBarFiller />
-      <Divider />
-      {props.drawerItems?.map((group, indexG) => {
-        if (group.drawer === false) return;
+  const handleButtonClick = () => {
+    if (drawerVariant === 'temporary') {
+      set_openNavigationDrawer(false);
+    }
+  };
 
-        return (
-          <List key={indexG}>
-            {group.groupName}
-            {group.items &&
-              group.items.map((item: any, indexI: number) => {
-                return (
-                  <SLink
-                    to={`${group.path}/${item.path}${searchParams ? searchParams : ''}`}
-                    key={indexI}
-                    disabled={
-                      !item.element ||
-                      // @ts-ignore
-                      (item.loginNeeded && props?.drawerLoginState?.[item.loginNeeded as 'web3' | 'node'] === false)
-                    }
-                  >
-                    <ListItem disablePadding>
-                      <ListItemButton
-                        disabled={
-                          !item.element ||
-                          // @ts-ignore
-                          (item.loginNeeded && props?.drawerLoginState?.[item.loginNeeded as 'web3' | 'node'] === false)
-                        }
-                      >
-                        <ListItemIcon>{item.icon}</ListItemIcon>
-                        <ListItemText primary={item.name} />
-                      </ListItemButton>
-                    </ListItem>
-                  </SLink>
-                );
-              })}
+  return (
+    <StyledDrawer
+      variant={drawerVariant}
+      open={openNavigationDrawer}
+      onClose={() => set_openNavigationDrawer(false)}
+    >
+      {drawerItems.map((group) => (
+        <div key={group.groupName}>
+          <Divider />
+          <List
+            subheader={
+              openNavigationDrawer ? (
+                <StyledListSubheader>{group.groupName}</StyledListSubheader>
+              ) : (
+                <Tooltip
+                  title={`Group: ${group.groupName.toLowerCase()}`}
+                  placement="right"
+                >
+                  <StyledListSubheader>
+                    <ListItemIcon sx={{ color: '#ddd' }}>{group.icon}</ListItemIcon>
+                  </StyledListSubheader>
+                </Tooltip>
+              )
+            }
+          >
+            {group.items.map((item) => (
+              <Tooltip
+                key={item.name}
+                title={!openNavigationDrawer && item.name}
+                placement="right"
+              >
+                <ListItemButton
+                  component={Link}
+                  to={`${group.path}/${item.path}${searchParams ?? ''}`}
+                  selected={location.pathname === `/${group.path}/${item.path}`}
+                  disabled={item.loginNeeded && !drawerLoginState?.[item.loginNeeded]}
```

Comment:
if a page should be disabled, it is avalible and if we click on it, the app crashes